Python/ibmcloudsql/cos.py

```diff
@@ -0,0 +1,659 @@
+        return self._get_default_cos_client(endpoint)
+
+    def _get_default_cos_client(self, endpoint=None):
+        '''
+        Create a low-level service client of COS from the default session
+        '''
+        def _get_default_s3_client(endpoint=None):
+            '''
+            Create a low-level service client of COS from the default session
+            '''
+            if endpoint is None:
+                endpoint = self.endpoint
+            assert (endpoint is not None)
+            return ibm_boto3.client('s3',
+                                    config=Config(signature_version='oauth'),
+                                    endpoint_url='https://{}'.format(endpoint))
+
+        return _get_default_s3_client(endpoint)
+
+    def list_cos_objects(self, cos_url):
+        """
+        List all objects in the current COS URL.
+
+        Parameters
+        ------------
+        url: str
+            A URI prefix. e.g., cos://us-south/<bucket-name>/object_path/
+
+        Returns
+        ----------
+        pd.DataFrame
+            The following columns ("Object", "Size", "StorageClass")
+
+        Note
+        ------
+        Having a trailing slash (/) makes a difference in the returned result, as an asterisk is added at the end. So,
+        "/prefix" would consider things like "/prefix_unexpected/value" and "/prefix/expected_value"; while
+        "/prefix/" only consider "/prefix/expected_value".
+
+        Examples
+        ------------
+            Only "list_objects' work for IBM COS, not 'list_objects_v2'
+
+            `IBM-COS-SDK <https://ibm.github.io/ibm-cos-sdk-python/reference/services/s3.html#S3.Client.list_objects_v2>`_
+        """
+        self.logon()
+
+        if cos_url[-1] != '/': 
+            cos_url = cos_url + '/'
+        cos_url = self.get_exact_url(cos_url)
+        url_parsed = self.analyze_cos_url(cos_url)
+        cos_client = self._get_default_cos_client(url_parsed.endpoint)
+        paginator = cos_client.get_paginator("list_objects")
+        page_iterator = paginator.paginate(Bucket=url_parsed.bucket,
+                                           Prefix=url_parsed.prefix)
+
+        result = None
+        for page in page_iterator:
+            if "Contents" in page:
+                #print(page['Contents'])
+                page_df = pd.DataFrame.from_dict(page['Contents'],
+                                                 orient='columns')
+                if result is None:
+                    result = page_df
+                else:
+                    result = result.append(page_df, sort=False)
+        if result is not None:
+            result = result.drop(columns=['ETag', 'Owner']).rename(
+                columns={"Key": "Object"})
+        else:
+            result = pd.DataFrame()
+            # result = pd.DataFrame(columns=['Object', 'LastModified', 'Size', 'StorageClass'])
+        return result
+
+    def delete_zero_size_objects(self, cos_url):
```

Comment:
delete_empty_objects

Reply:
Updated in commit `5cd3bc867c838a01f15f107bd7feac220ec15cfa`